src/LibraryManager/Providers/Unpkg/UnpkgCatalog.cs

```diff
@@ -170,7 +170,8 @@ public async Task<CompletionSet> GetLibraryCompletionSetAsync(string libraryName
 
             // Typing '@' after the library name should have version completion.
             int at = name.LastIndexOf('@');
-            name = at > -1 ? name.Substring(0, at) : name;
+            bool isScopedName = libraryNameStart.StartsWith("@", StringComparison.Ordinal);
```

Comment:
It feels like if we've already converted the library Id to `(name, version)` we shouldn't have to check for scopes again.  I think the original code here was handling a weird case from VersionedLibraryNamingScheme, in that for a partial name e.g. `jquery@` it would return `("jquery@", "")`.  

Note the @ is included in the name *only when there is no version segment*, whereas in a case where there is any version portion, it skips the at: jquery@1 would return `("jquery", "1")` and the `@` is dropped from both (this also grants a fun check - if `name.Length + version.Length == libraryNameStart.Length` then VLNS didn't skip over the @ sign and considers it a partial match.

I think there's a better ways to address this than re-dealing with scopes here.
1.  Change VLNS to not leave the @ appended in this "partial match" case.  I don't know why it does this, but it's just causing more work elsewhere (namely here and in JsDelivrCatalog).  I don't know if this behavior is relied upon elsewhere, we would need to check other uses of VLNS to see if there's any other paths where this might change a decision branch.
2. Do a smarter check instead of IndexOf tricks.  We're just looking to eliminate @ from the end, so:
```
string nonPartialName = name.TrimEnd("@");
if (caretPosition < name.Length + 1) // or caretPosition <= name.Length
```
should do the trick.

The JSDelivrCatalog has the same idea, but so much more succinct.  This one's a little harder to read, and I don't think it's correct when the caret position is not at the end of the `libraryNameStart`, but it probably works in the common case (though we should probably make them the same).

    if (caretPosition < name.Length + 1 && name[name.Length - 1] != '@')